saleor/app/tasks.py

```diff
@@ -21,13 +21,25 @@ def install_app_task(job_id, activate=False):
     except ValidationError as e:
         msg = ", ".join([f"{name}: {err}" for name, err in e.message_dict.items()])
         app_installation.message = msg
-    except (RequestException, HTTPError) as e:
-        logger.warning("Failed to install an app. error: %s", e)
+    except AppInstallationError as e:
+        logger.warning("Failed to install app. Error: %s", e)
+        app_installation.set_message(str(e))
+    except HTTPError as e:
+        logger.warning(
+            "Failed to install app. Response structure incorrect: missing error field."
+            " Error: %s",
+            e,
+        )
+        app_installation.message = (
+            "App internal error. Try later or contact with app support."
+        )
+    except RequestException as e:
+        logger.warning("Failed to install app. Error: %s", e)
         app_installation.message = (
             "Failed to connect to app. Try later or contact with app support."
         )
     except Exception as e:
-        logger.warning("Failed to install app. error %s", e)
+        logger.warning("Failed to install app. Error: %s", e)
         app_installation.message = "Unknown error. Contact with app support."
```

Comment:
> When building an app with a local Saleor instance, I would be happy to get here warning info like "response structure is incorrect: missing error field", something that would give me a more context of what is wrong or what is missing.

@korycins I refactored how installation errors are logged and added the "missing error field" hint.

Reply:
IMO *depending* on the error message in the response is not the way to go. It's not part of the HTTP spec ;). 
If installation failed because of some infrastructure _before_ the app (tunnel, proxy, etc), the message is not controllable by the app developer. In case of critical failures (500), handling them can also be out of dev control.

My suggestion is to:
- if the message in the response is present, use it
- if not, add a message with a response code

Later we can refactor it to put the response code in the context, but this solution will make debugging much more straightforward.

Reply:
I agree. I can change the error message to something like this:
```
App internal error (500). Try later or contact with app support.
```
Later we could simplify these messages and move error details to a dedicated place. Is that ok Krzych?

Reply:
Yup! I think that will make debugging easier until we settle down with the more complex `context`.